django_rest_passwordreset/views.py

```diff
@@ -119,10 +119,15 @@ def post(self, request, *args, **kwargs):
 
         # No active user found, raise a validation error
         if not active_user_found:
-            raise exceptions.ValidationError({
-                'email': [_(
-                    "There is no active user associated with this e-mail address or the password can not be changed")],
-            })
+            # if we dont want it to be known whether or not the email account
+            # exists in the db then we just return 200
+            if getattr(settings, 'DJANGO_REST_PASSWORDRESET_NO_INFORMATION_LEAKAGE', False):
```

Comment:
I think it would be better if we rewrite this

```python
        # No active user found, raise a validation error
        if not active_user_found:
            raise exceptions.ValidationError({
                'email': [_(
                    "There is no active user associated with this e-mail address or the password can not be changed")],
            })
```

to
```python
        # No active user found, raise a validation error
        if not active_user_found and not getattr(settings, 'DJANGO_REST_PASSWORDRESET_NO_INFORMATION_LEAKAGE', False):
            raise exceptions.ValidationError({
                'email': [_(
                    "There is no active user associated with this e-mail address or the password can not be changed")],
            })
```

as this would allow us to continue with the remaining code (the for loop will be ignored if there are no users in there...).

Reply:
sure thing